lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -1550,6 +1558,83 @@ void PreCheckExpression::resolveKeyPathExpr(KeyPathExpr *KPE) {
   KPE->resolveComponents(TC.Context, components);
 }
 
+Expr *PreCheckExpression::simplifyTypeConstructionWithLiteralArg(Expr *E) {
+  auto *call = dyn_cast<CallExpr>(E);
+  if (!call || call->getNumArguments() != 1)
+    return nullptr;
+
+  auto *typeExpr = dyn_cast<TypeExpr>(call->getFn());
+  if (!typeExpr)
+    return nullptr;
+
+  auto *argExpr = call->getArg()->getSemanticsProvidingExpr();
+  auto *literal = dyn_cast<LiteralExpr>(argExpr);
+  if (!literal)
+    return nullptr;
+
+  auto *protocol = TC.getLiteralProtocol(literal);
+  if (!protocol)
+    return nullptr;
+
+  Type type;
+  if (auto *rep = typeExpr->getTypeRepr()) {
+    TypeResolutionOptions options;
+    options |= TypeResolutionFlags::AllowUnboundGenerics;
+    options |= TypeResolutionFlags::InExpression;
+    type = TC.resolveType(rep, DC, options);
+  } else {
+    type = typeExpr->getTypeLoc().getType();
+  }
+
+  // If type is unknown, or this is an explicit call to failable init.
+  if (!type || type->getOptionalObjectType())
+    return nullptr;
+
+  ConformanceCheckOptions options;
+  options |= ConformanceCheckFlags::InExpression;
+  options |= ConformanceCheckFlags::SuppressDependencyTracking;
```

Comment:
Is it really correct to suppress dependency tracking?

Can you test what happens if a type conforms to a literal protocol conditionally?

Reply:
Sure, I can try to do that, @DougGregor mentioned to me before that it should be suppressed...

Reply:
Actually next flag skip conditional conformances so I think it should be ok to skip dependency tracking?

Reply:
Dependency tracking is about incremental builds -- it's not related to conditional conformances. @jrose-apple will better explain what it's for

Reply:
I just thought that two questions are related...

Reply:
!! It is not okay to suppress dependency tracking, please do not do this!